CMakeLists.txt

```diff
@@ -51,9 +51,13 @@ ADD_LIBRARY(hiredis::hiredis ALIAS hiredis)
 set(hiredis_export_name hiredis CACHE STRING "Name of the exported target")
 set_target_properties(hiredis PROPERTIES EXPORT_NAME ${hiredis_export_name})
 
+# Extract the first part of the HIREDIS_SONAME (major version) and use it for SOVERSION
+STRING(REGEX REPLACE "^([0-9]+)\\..*" "\\1" HIREDIS_SONAME_MAJOR "${HIREDIS_SONAME}")
```

Comment:
Maybe we actually drop `HIREDIS_SONAME` from `hiredi.h` and just use `${HIREDIS_MAJOR}` in here where appropriate?